### PR TITLE
Issues

### LINKED ISSUE #196: [Feature Request] ARC.init needs .gitignore with WRITE contract
Should contain default ignore information

### LINKED ISSUE #206: [Feature Request] Unify IO function names accross xlsx and json

### LINKED ISSUE #208: [Feature Request] UpdateFsByISA on ARC.ISA setter

### LINKED ISSUE #214: [BUG?] Wrong regex parsing for term headers
```ps
ERROR: Error loading template /dataplant/MIAPPE_nutrients.xlsx: Could not parse header group [L1 : Parameter [Concentration of [nutrient] before start of the experiment] | String; M1 : Term Source REF (MIAPPE:0134) | String; N1 : Term Accession Number (MIAPPE:0134) | String]
```

### LINKED ISSUE #215: [Feature Request] publish npm misses index.js

### LINKED ISSUE #219: [Feature Request] extend structural equality.
- ArcAssay
- ArcStudy
- ArcInvestigation

### PR DESCRIPTION
**Pull request by Freymaurer**
- closes #215 
- closes #214 
- closes #208
- closes #206 
- closes #196
- closes #219

This PR closes milestone https://github.com/nfdi4plants/ARCtrl/milestone/6.

### CODE REVIEW

File: src/ARCtrl/ARCtrl.fs
Comment (HLWeil) [inline src/ARCtrl/ARCtrl.fs:1]: btw rename this file to ARC.fs?

File: src/ISA/ISA.Json/Assay.fs [hunk 1]
```diff
@@ -71,10 +71,10 @@ module Assay =
             }
         )
 
-    let fromString (s:string) = 
-        GDecode.fromString (decoder (ConverterOptions())) s
+    let decodeFromString (s:string) =
```
Comment (HLWeil) [inline src/ISA/ISA.Json/Assay.fs:74]: @kMutagene
What do you think about this function naming?

The function has a json-string as input and returns the decoded object.

File: src/ISA/ISA/Regex.fs [hunk 2]
```diff
@@ -96,7 +96,7 @@ module Pattern =
     ///
     /// Exmp. 2: Characteristic [species] --> termcolumntype: Characteristic; termname: species
     [<LiteralAttribute>]
-    let TermColumnPattern = @"(?<termcolumntype>.+)\s\[(?<termname>.+)\]"
+    let TermColumnPattern = @"(?<termcolumntype>.+?)\s\[(?<termname>.+)\]"
```
Comment (HLWeil) [inline src/ISA/ISA/Regex.fs:99]: Crazy that this already does the trick

File: src/ARCtrl/ARCtrl.fs [hunk 3]
```diff
@@ -355,6 +356,9 @@ type ARC(?isa : ISA.ArcInvestigation, ?cwl : CWL.CWL, ?fs : FileSystem.FileSyste
         |> Option.bind (fun tree -> if ignoreHidden then tree |> FileSystemTree.filterFolders (fun n -> not (n.StartsWith("."))) else Some tree)
         |> Option.defaultValue (FileSystemTree.fromFilePaths [||])
         
+    static member DefaultContracts = Map<string,Contract> [|
```
Comment (HLWeil) [inline src/ARCtrl/ARCtrl.fs:359]: Needed in some function or just for user convenience? And if so, why map when the Contract does contain the path itself?

**Review by HLWeil — CHANGES_REQUESTED** [review]
https://github.com/nfdi4plants/ARCtrl/issues/206 meant unification between xlsx and json. Now we have
```fsharp
ARCtrl.ISA.Spreadsheet.ArcInvestigation.fromFsWorkbook
```
and 
```fsharp
ARCtrl.ISA.Json.ArcInvestigation.decodeFromString
```
😅 
Or is this fine, I don't know

File: src/ARCtrl/ARCtrl.fs (hunk 3 above)
Comment (Freymaurer) [reply src/ARCtrl/ARCtrl.fs:359]: This is api user convinience and a map so the contract is easier to find? That part is up for discussion

File: src/ISA/ISA.Json/Assay.fs (hunk 1 above)
Comment (kMutagene) [reply src/ISA/ISA.Json/Assay.fs:74]: is "decode" a toth thing? nonetheless, people usually expect this to be called deserialize

**Review by HLWeil — APPROVED** [review]

**Pull request merged**